torchrl/data/tensor_specs.py

```diff
@@ -1456,7 +1456,7 @@ def is_in(self, val: torch.Tensor) -> bool:
         shape = torch.broadcast_shapes(shape, val.shape)
         mask_expand = self.mask.expand(shape)
         gathered = mask_expand & val
-        return gathered.any(-1).all()
+        return gathered.any(-1).all() and val.dtype == self.dtype
```

Comment:
this isn't related to this PR but I think the proper check should be

```suggestion
        return (gathered.sum(-1) == 1).all() and val.dtype == self.dtype
```

Reply:
Should I add this to the current PR?

Reply:
Yes I think those two are related and can be part of the same PR

Reply:
Okay, done